### ISSUE TITLE
Incorrect appearance of useless-param-doc warnings.

### ISSUE THREAD

**Issue by KyleMylonakisProtopia** [PyCQA/pylint#8421]
### Bug description

Recently, functions in our codebase have been flagged with the pylint warning `useless-param-doc`

Here is an example which triggers `"c, d, e" useless ignored parameter documentation Pylint(W9019:useless-param-doc)`

```python
def min_example(a: str, b: str, c: List[str], d: str, e: bool = False) -> None:
    """Perform a task.

    Supports arbitrary length lists.

    Args:
        a: The name.
        b: The location.
        c: A list of.
        d: The path to.
        e: Whether.

    Raises:
        RuntimeError: Raised if.
        ValueError: If.
    """
```

### Command used

```shell
`pylint <FILE>` and built in linter to VS Code
```

### Pylint output

```shell
"c, d, e" useless ignored parameter documentation Pylint(W9019:useless-param-doc)
```

### Expected behavior

There should be no warning

### Pylint version

```shell
pylint 2.17.0
astroid 2.15.0
Python 3.10.8 (main, Nov 24 2022, 08:08:27) [Clang 14.0.6 ]
```

### OS / Environment

Mac OS 13.2.1 and Ubuntu with linux kernel 5.15.0-60-generic.

**Comment by mbyrnepr2** [comment]
Thanks @KyleMylonakisProtopia. I need to tweak the parameter-naming before in order to reproduce this; is the code in the description an exact reproducer for you?

```python
from typing import List

# IGNORED_ARGUMENT_NAMES = re.compile("_.*|^ignored_|^unused_")
# I.e. names prefixed with:
# _*
# ignored_
# unused_


def min_example(a: str, b: str, ignored_c: List[str], unused_d: str, _e: bool = False) -> None:
    """Perform a task.

    Supports arbitrary length lists.

    Args:
        a: The name.
        b: The location.
        ignored_c: A list of.
        unused_d: The path to.
        _e: Whether.

    Raises:
        RuntimeError: Raised if.
        ValueError: If.
    """
```

**Comment by KyleMylonakisProtopia** [comment]
For some reason neither of these examples is triggering the useless-param-doc warning anymore on this reported system.

Does your provided MWE still trigger this warning on your system?

**Comment by mbyrnepr2** [comment]
Yes; my example above does still trigger the message, as expected. It is reproduced by loading the extension and running Pylint on the module with the example code:
```python
pylint example.py --load-extensions=pylint.extensions.docparams
example.py:10:0: W9019: "_e, ignored_c, unused_d" useless ignored parameter documentation (useless-param-doc)
```

Perhaps the extension is not getting loaded in your case?

**Comment by KyleMylonakisProtopia** [comment]
Whoops, I noticed that the `pyproject.toml` file disabled the error (I guess someone added this between me making this ticket and commenting the other day). 

When I commented this disable flag out, the minimum working examples provided in this thread triggered the warning!

This is in fact reproducible on my end.

**Comment by mbyrnepr2** [comment]
Re-opened. Just to confirm - you are able to reproduce even when the arguments are not prefixed with a `_`, `ignored_` or `unused_`?

**Comment by KyleMylonakisProtopia** [comment]
Correct.

**Comment by mbyrnepr2** [comment]
@KyleMylonakisProtopia could you please check some files which your project may contain with Pylint config? For example the pyproject.toml, tox.ini, setup.cfg etc.? I can only see a way of reproducing with the original code in the description, if `c,d and e` are listed as `ignored-argument-names` in a configuration somewhere. From the commandline it would look like this: `pylint example.py --load-extensions=pylint.extensions.docparams --ignored-argument-names=[c,d,e]`.

Perhaps as a sanity-check you could try the code from the description with completely different argument-names - to be sure they are not listed as ignored arguments in configuration somewhere; it shouldn't emit the message if you use `pylint example.py --load-extensions=pylint.extensions.docparams` only.

**Comment by KyleMylonakisProtopia** [comment]
I can change the variable names in the MWE again, but as you may expect, but we see this all over our codebase with different variable names. Other than that we have various seemingly unrelated settings and load the google style plugin.

Here is our `ignored-argument-names` in our `pyproject.toml` file.

```toml
ignored-argument-names = "^[_|ignored_|unused_|mocks|mock_]"
```

Is that enough for reproduction?

**Comment by mbyrnepr2** [comment]
That regex is the problem I think @KyleMylonakisProtopia! the [square braces ](https://docs.python.org/3/library/re.html#regular-expression-syntax)are capturing a set of characters.
In the case of `"^[_|ignored_|unused_|mocks|mock_]"`, this is also matching on `c`, `d` and `e` since they are contained in the set (but not `a` and `b` as in your original code example).

```python
>>>import re
>>>pattern = "^[_|ignored_|unused_|mocks|mock_]"
>>> re.match(p, 'a')
>>> re.match(p, 'b')
>>> re.match(p, 'c')
<re.Match object; span=(0, 1), match='c'>
>>> re.match(p, 'd')
<re.Match object; span=(0, 1), match='d'>
>>> re.match(p, 'e')
<re.Match object; span=(0, 1), match='e'>
```

**Comment by KyleMylonakisProtopia** [comment]
That change resolved it! Thanks for the help!
